src/sb-util.ts

```diff
@@ -199,7 +196,8 @@ export class BlockCollection implements Queryable {
     }
 
     public first(): Block {
-        return first(makeIterable(this, (): Iterator<Block> => this[Symbol.iterator]()));
+        const iter = (): Iterator<Block> => this[Symbol.iterator]();
+        return first(new MemoizedIterable(iter()));
```

Comment:
any reason this can't just be `first(this[Symbol.iterator]())` ?

Reply:
this change could also be moved into it's own small PR (please do since it wouldn't have anything to do with Memoized)